waspc/cli/Command/Telemetry/Project.hs

```diff
@@ -89,17 +92,31 @@ initialCache = ProjectTelemetryCache { _lastCheckIn = Nothing, _lastCheckInBuild
 
 -- * Project telemetry cache file.
 
-readOrCreateProjectTelemetryFile :: Path Abs (Dir TelemetryCacheDir) -> ProjectHash -> IO ProjectTelemetryCache
-readOrCreateProjectTelemetryFile telemetryCacheDirPath projectHash = do
-    fileExists <- SD.doesFileExist filePathFP
-    maybeCache <- if fileExists then readCacheFile else return Nothing
-    case maybeCache of
-        Just cache -> return cache
-        Nothing -> writeProjectTelemetryFile telemetryCacheDirPath projectHash initialCache >> return initialCache
+convertMaybeTimeToString :: Maybe T.UTCTime -> String
+convertMaybeTimeToString = maybe "nil" show
+
+showLastSentTelemetryDateTime :: Maybe ProjectTelemetryCache -> String
+showLastSentTelemetryDateTime maybeCache = case maybeCache of
+    Just cache -> "Telemetry data was last sent at: " <> convertMaybeTimeToString (_lastCheckIn cache) <>
+                  "\nTelemetry data was last upon using `build` was at: " <> convertMaybeTimeToString (_lastCheckInBuild cache)
+    Nothing    -> "Failed to find/read the telemetry file to display sufficient information"
+
+readProjectTelemetryFile :: Path Abs (Dir TelemetryCacheDir) -> ProjectHash -> Maybe ProjectTelemetryCache
+readProjectTelemetryFile telemetryCacheDirPath projectHash = do
+    fileExists <- liftIO $ SD.doesFileExist filePathFP
+    join (if fileExists then liftIO readCacheFile else Nothing)
   where
       filePathFP = SP.toFilePath $ getProjectTelemetryFilePath telemetryCacheDirPath projectHash
       readCacheFile = Aeson.decode . ByteStringLazyUTF8.fromString <$> readFile filePathFP
 
+readOrCreateProjectTelemetryFile :: Path Abs (Dir TelemetryCacheDir) -> ProjectHash -> IO ProjectTelemetryCache
+readOrCreateProjectTelemetryFile telemetryCacheDirPath projectHash = do
```

Comment:
do block + let-in immediately in it -> something is redundant here, right? I would just go with `do` and perform the `let` as part of the `do` sugar, I wouldn't create new `let-in` expression.